src/linear_algebra/dual.lean

```diff
@@ -558,3 +558,81 @@ end
 end
 
 end subspace
+
+namespace linear_map
+
+variables {R : Type*} [comm_ring R] {M₁ : Type*} {M₂ : Type*}
+variables [add_comm_group M₁] [module R M₁] [add_comm_group M₂] [module R M₂]
+
+open module
+
+/-- Given a linear map `f : M₁ →ₗ[R] M₂`, `f.dual_map` is the linear map between the dual of
+`M₂` and `M₁` such that it maps the functional `φ` to `φ ∘ f`. -/
+def dual_map (f : M₁ →ₗ[R] M₂) : dual R M₂ →ₗ[R] dual R M₁ :=
+lcomp R R f
+
+@[simp] lemma dual_map_apply (f : M₁ →ₗ[R] M₂) (g : dual R M₂) (x : M₁) :
+  f.dual_map g x = g (f x) :=
+lcomp_apply f g x
+
+variable {f : M₁ →ₗ[R] M₂}
+
+lemma ker_dual_map_eq_range_dual_annihilator :
+  f.dual_map.ker = f.range.dual_annihilator :=
+begin
+  ext φ, split; intro hφ,
+  { rw mem_ker at hφ,
+    rw submodule.mem_dual_annihilator,
+    rintro y ⟨x, _, rfl⟩,
+    rw [← dual_map_apply, hφ],
+    refl },
+  { ext x,
+    rw dual_map_apply,
+    rw submodule.mem_dual_annihilator at hφ,
+    exact hφ (f x) ⟨x,
+      show x ∈ (⊤ : submodule R M₁), by exact submodule.mem_top, rfl⟩ }
```

Comment:
This worked without the `show` for me.

Reply:
Without the `show` its giving me the error `don't know how to synthesize placeholder`

Reply:
Hmm, perhaps `gitpod` is not to be trusted. Does it work as `submodule.coe_mem.2 submodule.mem_top` or similar?

Reply:
Works after I add a underscore, i.e. `(submodule.mem_coe _).mpr submodule.mem_top`. Will change to this